src/data_info/services/metadata.clj

```diff
@@ -199,9 +196,10 @@
    :type (resolve-data-type type)})
 
 (defn- get-writable-data-items
+  "Gets the path, type, and id for data-ids and validates all the paths are writeable."
   [cm user data-ids]
   (validators/validate-num-paths data-ids)
-  (let [data-items (map (partial stat/uuid-stat cm user) data-ids)
+  (let [data-items (map #(stat/uuid-stat cm user % :filter-include [:path :type :id]) data-ids)
```

Comment:
This filter down to this is based on what's used in the one calling location, `metadata-copy`, below. The maps after it in the function get the path and the ID, and then the format-copy-dest-item function uses the ID and type.